graphrag/index/__main__.py

```diff
@@ -4,73 +4,123 @@
 """The Indexing Engine package root."""
 
 import argparse
+from enum import Enum
+from pathlib import Path
 
 from .cli import index_cli
 
+
+class ReporterType(Enum):
+    """The type of reporter to use."""
+
+    RICH = "rich"
```

Comment:
I've always wanted to replace our enum strings with auto(), would make the implementation cleaner.
WDYT? Not suggesting this as a refactor, but a conversation starter.

Reply:
BTW, fully aware this is a cli PR :) I'm speaking in general hehe

Reply:
Sounds like a good idea initially however I think we need the string representation for the CLI. I don't think `argparse` can display/handle enums very cleanly without an overloaded `__str__` or requiring more code.

For internal library operations, `auto` would be cleaner but then we're maintaining two different styles of enums which is worse in my opinion ☹️.